### PR TITLE
Create type definition for synaptic

### PR DESCRIPTION
**Pull request by ToastHawaii**
Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

### CODE REVIEW

**Comment by vvakame** [comment]
@zhengbli What's mean `Revision needed` in this case?

**Review by zhengbli — CHANGES_REQUESTED** [review]

File: synaptic/synaptic.d.ts [hunk 1]
```diff
@@ -0,0 +1,769 @@
...
+         */
+        public project(otherNetwork: Network | Layer, connectionType: Layer.connectionType): Layer.LayerConnection;
+
+        /**
+         * A network can project a connection to another, or gate a connection between two others networks in the same way Layers do.
+         * @param otherNetwork
+         * @param connectionType
+         * @param weights
+         */
+        public project(otherNetwork: Network | Layer, connectionType: Layer.connectionType, weights: any): Layer.LayerConnection;
```
Comment (ToastHawaii) [inline synaptic/synaptic.d.ts]: done

File: synaptic/synaptic.d.ts [hunk 2]
```diff
@@ -0,0 +1,769 @@
...
+        /**
+         * This method trains an XOR to the network, is useful when you are experimenting with different architectures and you want to test and compare their performances.
+         */
+        public XOR(): Trainer.TrainingResult;
+
+        /**
+         * This method trains an XOR to the network, is useful when you are experimenting with different architectures and you want to test and compare their performances.
+         * @param options
+         */
+        public XOR(options: Trainer.XOROptions): Trainer.TrainingResult;
```
Comment (ToastHawaii) [inline synaptic/synaptic.d.ts]: done

File: synaptic/synaptic.d.ts [hunk 3]
```diff
@@ -0,0 +1,769 @@
...
+        /**
+         * This method trains the network to pass an Embedded Reber Grammar test.
+         */
+        public ERG(): Trainer.ERGTrainingResult;
+
+        /**
+         * This method trains the network to pass an Embedded Reber Grammar test.
+         * @param options
+         */
+        public ERG(options: Trainer.ERGOptions): Trainer.ERGTrainingResult;
```
Comment (ToastHawaii) [inline synaptic/synaptic.d.ts]: done

**Pull request merged**